### PR TITLE
Factorize handling of control flags between synterp and interp + use control flags in stm close_proof calls

### PR DESCRIPTION
**Pull request by SkySkimmer**
The main goal is to include the close_proof time in
Time/Instructions/Timeout results.

### CODE REVIEW

File: vernac/vernacinterp.ml [hunk 1]
```diff
@@ -13,89 +13,49 @@ open Synterp
...
+  let noop = st.Vernacstate.interp.lemmas, st.Vernacstate.interp.program in
+  let control, res =
+    VernacControl.under_control ~loc
+      ~with_local_state:(with_interp_state ~unfreeze_transient st)
+      control
+      ~noop
+      f
+  in
+  if VernacControl.after_last_phase ~loc control
+  then noop
```
Comment (SkySkimmer) [inline vernac/vernacinterp.ml:43]: the handling of unchanged functionally passed state here feels a bit awkward so maybe the api could be improved a bit
Not sure how though

**Comment by SkySkimmer** [comment]
How Redirect should be handled when the phase split is used is not fully clear. 

In master any synterp output will be deleted when the file is reopened for the interp phase (even if the interp phase is noop).

In this PR the synterp phase will delete whatever is in the file before writing its output, and the interp phase wll append its output to whatever is there. This means if the IDE runs the interp phase twice without rerunning the synterp phase, we will get the interp output twice.

**Comment by SkySkimmer** [comment]
ping @coq/vernac-maintainers

File: stm/stm.ml [hunk 2]
```diff
@@ -2097,12 +2097,21 @@ let known_state ~doc ?(redefine_qed=false) ~cache id =
...
+                    let control = VernacControl.from_syntax control in
+                    let control, proof =
+                      VernacControl.under_control ~loc:x.expr.loc
+                        ~with_local_state:VernacControl.trivial_state
+                        control
+                        ~noop:None
+                        (fun () -> Some (PG_compat.close_future_proof ~feedback_id:id fp'))
+                    in
+                    (* We only get [noop] from [Fail],
+                       but we can't see [Fail] in this classification. *)
```
Comment (gares) [inline stm/stm.ml:2110]: I don't get this message. Is noop actually None?

Comment (gares) [reply stm/stm.ml:2110]: I think I see now, but then I believe it could happen.

Comment (SkySkimmer) [reply stm/stm.ml:2110]: How could it happen?

Comment (gares) [reply stm/stm.ml:2110]: Maybe I'm mistaken, but this is what I would try.
```
Lemma x : True.
Proof.
trivial.
Fail Qed.
```
If it works, just ignore my remark.

Comment (SkySkimmer) [reply stm/stm.ml:2110]: It works
It also works with no `trivial` (which is the noop case)

**Review by gares — APPROVED** [review]

**Comment by SkySkimmer** [comment]
merge?

**Pull request merged**